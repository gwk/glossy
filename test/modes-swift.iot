{
  'compile': ['$PROJ/legs/__main__.py test/modes.legs -test -output $NAME.swift'],
  'compile_timeout': 120,
  'cmd': './$NAME.swift',
  'args': [
    'a (bc def) g',
    '/* comment "contents" () / *. */',
    "'single quoted.' \"double quoted.\"",
    '/* nested /* comment. */ */',
    "'sq \(a) interpolated.'",
    '"dq \(a) interpolated."',
    "'sq \(nested \"interpolated \(a)\").'",
  ],
}