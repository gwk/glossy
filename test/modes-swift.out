
arg1: "a (bc def) g"
token: arg1:1:1-2: sym
  a (bc def) g
  ~
token: arg1:1:2-3: space
  a (bc def) g
   ~
token: arg1:1:3-4: `(`
  a (bc def) g
    ~
token: arg1:1:4-6: sym
  a (bc def) g
     ~~
token: arg1:1:6-7: space
  a (bc def) g
       ~
token: arg1:1:7-10: sym
  a (bc def) g
        ~~~
token: arg1:1:10-11: `)`
  a (bc def) g
           ~
token: arg1:1:11-12: space
  a (bc def) g
            ~
token: arg1:1:12-13: sym
  a (bc def) g
             ~

arg2: "/* comment \"contents\" () / *. */"
token: arg2:1:1-3: `/*`
  /* comment "contents" () / *. */
  ~~
token: arg2:1:3-26: comment_contents
  /* comment "contents" () / *. */
    ~~~~~~~~~~~~~~~~~~~~~~~
token: arg2:1:26-27: `/`
  /* comment "contents" () / *. */
                           ~
token: arg2:1:27-28: comment_contents
  /* comment "contents" () / *. */
                            ~
token: arg2:1:28-29: `*`
  /* comment "contents" () / *. */
                             ~
token: arg2:1:29-31: comment_contents
  /* comment "contents" () / *. */
                              ~~
token: arg2:1:31-33: `*/`
  /* comment "contents" () / *. */
                                ~~

arg3: "'single quoted.' \"double quoted.\""
token: arg3:1:1-2: `'`
  'single quoted.' "double quoted."
  ~
token: arg3:1:2-16: lit_contents
  'single quoted.' "double quoted."
   ~~~~~~~~~~~~~~
token: arg3:1:16-17: `'`
  'single quoted.' "double quoted."
                 ~
token: arg3:1:17-18: space
  'single quoted.' "double quoted."
                  ~
token: arg3:1:18-19: `"`
  'single quoted.' "double quoted."
                   ~
token: arg3:1:19-33: lit_contents
  'single quoted.' "double quoted."
                    ~~~~~~~~~~~~~~
token: arg3:1:33-34: `"`
  'single quoted.' "double quoted."
                                  ~

arg4: "/* nested /* comment. */ */"
token: arg4:1:1-3: `/*`
  /* nested /* comment. */ */
  ~~
token: arg4:1:3-11: comment_contents
  /* nested /* comment. */ */
    ~~~~~~~~
token: arg4:1:11-13: `/*`
  /* nested /* comment. */ */
            ~~
token: arg4:1:13-23: comment_contents
  /* nested /* comment. */ */
              ~~~~~~~~~~
token: arg4:1:23-25: `*/`
  /* nested /* comment. */ */
                        ~~
token: arg4:1:25-26: comment_contents
  /* nested /* comment. */ */
                          ~
token: arg4:1:26-28: `*/`
  /* nested /* comment. */ */
                           ~~

arg5: "'sq \\(a) interpolated.'"
token: arg5:1:1-2: `'`
  'sq \(a) interpolated.'
  ~
token: arg5:1:2-5: lit_contents
  'sq \(a) interpolated.'
   ~~~
token: arg5:1:5-7: `\(`
  'sq \(a) interpolated.'
      ~~
token: arg5:1:7-8: sym
  'sq \(a) interpolated.'
        ~
token: arg5:1:8-9: `)`
  'sq \(a) interpolated.'
         ~
token: arg5:1:9-23: lit_contents
  'sq \(a) interpolated.'
          ~~~~~~~~~~~~~~
token: arg5:1:23-24: `'`
  'sq \(a) interpolated.'
                        ~

arg6: "\"dq \\(a) interpolated.\""
token: arg6:1:1-2: `"`
  "dq \(a) interpolated."
  ~
token: arg6:1:2-5: lit_contents
  "dq \(a) interpolated."
   ~~~
token: arg6:1:5-7: `\(`
  "dq \(a) interpolated."
      ~~
token: arg6:1:7-8: sym
  "dq \(a) interpolated."
        ~
token: arg6:1:8-9: `)`
  "dq \(a) interpolated."
         ~
token: arg6:1:9-23: lit_contents
  "dq \(a) interpolated."
          ~~~~~~~~~~~~~~
token: arg6:1:23-24: `"`
  "dq \(a) interpolated."
                        ~

arg7: "'sq \\(nested \"interpolated \\(a)\").'"
token: arg7:1:1-2: `'`
  'sq \(nested "interpolated \(a)").'
  ~
token: arg7:1:2-5: lit_contents
  'sq \(nested "interpolated \(a)").'
   ~~~
token: arg7:1:5-7: `\(`
  'sq \(nested "interpolated \(a)").'
      ~~
token: arg7:1:7-13: sym
  'sq \(nested "interpolated \(a)").'
        ~~~~~~
token: arg7:1:13-14: space
  'sq \(nested "interpolated \(a)").'
              ~
token: arg7:1:14-15: `"`
  'sq \(nested "interpolated \(a)").'
               ~
token: arg7:1:15-28: lit_contents
  'sq \(nested "interpolated \(a)").'
                ~~~~~~~~~~~~~
token: arg7:1:28-30: `\(`
  'sq \(nested "interpolated \(a)").'
                             ~~
token: arg7:1:30-31: sym
  'sq \(nested "interpolated \(a)").'
                               ~
token: arg7:1:31-32: `)`
  'sq \(nested "interpolated \(a)").'
                                ~
token: arg7:1:32-33: `"`
  'sq \(nested "interpolated \(a)").'
                                 ~
token: arg7:1:33-34: `)`
  'sq \(nested "interpolated \(a)").'
                                  ~
token: arg7:1:34-35: lit_contents
  'sq \(nested "interpolated \(a)").'
                                   ~
token: arg7:1:35-36: `'`
  'sq \(nested "interpolated \(a)").'
                                    ~
